### PR TITLE
Feature: Allow passing axios instance or config in options.

### PR DESCRIPTION
**Pull request by krisdages**
### Background:

I have an Electron app using axios for HTTP where I am using a custom adapter to get around proxy issues by using Electron's `net` module.

### My workaround:
The current version of `analytics-node` doesn't expose any configuration for `axios`, so I added the following `options` properties:
- `axiosConfig`: configuration options for creating an axios instance if you don't have axios as a dependency
- `axiosInstance`: instance of axios if you already have axios as a dependency, maybe with a different version

It's not ideal from a library-coupling perspective, but I didn't see an easy way to avoid that without 
defining some library-agnostic interface for the HTTP adapter which would still support retries, which seems overkill.

I just thought I would post this PR in case it was a useful enhancement. 

### Alternative considered:
One option `axios` with either a config object or a function returning an axios instance.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,10 +21,12 @@ class Analytics {
...
    * @param {String} writeKey
    * @param {Object} [options] (optional)
-   *   @property {Number} flushAt (default: 20)
-   *   @property {Number} flushInterval (default: 10000)
-   *   @property {String} host (default: 'https://api.segment.io')
-   *   @property {Boolean} enable (default: true)
+   *   @property {Number} [flushAt] (default: 20)
+   *   @property {Number} [flushInterval] (default: 10000)
+   *   @property {String} [host] (default: 'https://api.segment.io')
+   *   @property {Boolean} [enable] (default: true)
```
Comment (pbassut) [inline index.js:27]: Thanks for those updates

**Review by pbassut — APPROVED** [review]
If we merge this we can work out https://github.com/segmentio/analytics-node/pull/246 by using axios' baseURL parameter(and also leaving feature like proxy intact).

We can merge this and I'll tackle that later. Leaving this here for reference

**Review by pooyaj — APPROVED** [review]

**Comment by pooyaj** [comment]
@krisdages @pbassut could you please fix the conflict so we can merge this?

**Comment by pbassut** [comment]
Sure. Notice we don't need to merge https://github.com/segmentio/analytics-node/pull/246 if we merge this.
I would even prefer to merge this as this provides more flexibility.

**Comment by pbassut** [comment]
Also, I'll have to create another PR to fix the conflicts here.

**Comment by pooyaj** [comment]
@pbassut oh, sorry just merged the other one. Agree, this would've covered that.

**Comment by pbassut** [comment]
https://github.com/segmentio/analytics-node/pull/270

**Pull request merged**